src/data/repositories/Dhis2DocumentRepository.ts

```diff
@@ -0,0 +1,99 @@
+import { createFFmpeg, fetchFile } from "@ffmpeg/ffmpeg";
+import FileType from "file-type/browser";
+import Resizer from "react-image-file-resizer";
+import { Document } from "../../domain/entities/Document";
+import { DocumentRepository } from "../../domain/repositories/DocumentRepository";
+import { D2Api } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../utils/d2-api";
+import { getUid } from "../utils/uid";
+import { Instance } from "../entities/Instance";
+
+export class Dhis2DocumentRepository implements DocumentRepository {
```

Comment:
@tokland this one was just moving the functions from InstanceRepository, it was Alexis's code, but I think it might be useful to review it if you think so because it could be taken later as the "example" for our other repositories or future ones. In my opinion I would only remove the ffmpeg part applying to gifs, but you will be able to review the code in a much deeper way if you think it could be relevant to review it

Reply:
It's not a too long class, it can be reviewed, yes. I don't have the context on why ffmpeg was used here, so I cannot say.